File: codeforces/587A/py/587a.py
import sys

n = int(input())
wn = list(map(int, sys.stdin.readline().split()))

counts = [0] * 1000101
for i in range(n):
    counts[wn[i]] += 1

ans = 0
for i in range(1000100):
    if counts[i] == 0:
        continue

    counts[i+1] += counts[i] >> 1
    if counts[i] & 1 != 0:
        ans += 1

print(ans)

File: codeforces/587A/cpp/587a.cpp
#include <cstdio>

int main()
{
    int n, x;
    scanf("%d", &n);

    int counts[1000100] = {0};
    for (int i = 0; i < n; i++) {
        scanf("%d", &x);
        counts[x]++;
    }

    int ans = 0;
    for (int i = 0; i < 1000100; i++) {
        if (counts[i] == 0) {
            continue;
        }

        counts[i+1] += counts[i] >> 1;
        if (counts[i] & 1) {
            ans++;
        }
    }

    printf("%d\n", ans);

    return 0;
}
